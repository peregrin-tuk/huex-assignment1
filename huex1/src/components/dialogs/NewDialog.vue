<template>
  <v-card>
    <v-card-title>Create new board</v-card-title>
    <v-card-text>
      Create a new board? You can reopen this board by clicking on the <strong>Open</strong> button on the left side, or by saving and opening the URL at the top of your address bar.
    </v-card-text>
    <v-card-actions>
      <v-spacer></v-spacer>
      <v-btn color="accent" text @click="$parent.$parent.$parent.closeDialog()">Cancel</v-btn>
      <v-btn color="primary" @click="addBoard">Create new board</v-btn>
    </v-card-actions>
  </v-card> 
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "OpenDialog",
  data: () => ({
    copied: false,
  }),
  methods: {
    addBoard: function() {
      this.$store.dispatch('addNewBoard')
      // TODO close dialog
    }
  }
});
</script>

<style lang="scss" scoped>
  .copy-button {
    top: -12px;
    width: 9rem;
  }
</style>