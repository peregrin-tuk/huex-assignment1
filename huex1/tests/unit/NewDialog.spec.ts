import {createLocalVue, shallowMount, mount} from '@vue/test-utils';
import NewDialog from "@/components/dialogs/NewDialog.vue";
// eslint-disable-next-line no-unused-vars
import Vuex, { Store } from 'vuex'
import Vuetify from 'vuetify';

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

describe('Toolbutton', () => {
  let actions: { addNewBoard: jest.Mock};
  let store: Store<any>;

  beforeEach(() => {
    actions = {
      addNewBoard: jest.fn()
    }
    store = new Vuex.Store({
      actions,
      state: {
        currentBoard: {
          name: 'Awesome name',
          key: '#123456789',
          content: [],
          id: '1234'
        }
      }
    })
  })

  it('should match snapshot', () => {
    const wrapper = shallowMount(NewDialog, { 
      store,
      localVue,
    });

    expect(wrapper.html()).toMatchSnapshot();
  })

  it('should render a title', () => {
    const wrapper = mount(NewDialog, { 
      store, 
      localVue,
    });

    const el = wrapper.find('.v-card__title');
    expect(el.text()).toBe('Create new board');
  })

  it('should render a card text', () => {
    const wrapper = mount(NewDialog, { 
      store, 
      localVue,
    });

    const el = wrapper.find('.v-card__text');
    expect(el.text()).toBe('Create a new board? You can reopen this board by clicking on the Open button on the left side, or by saving and opening the URL at the top of your address bar.');
  })

  it('should render a red cancel button and a green new button', () => {
    const wrapper = mount(NewDialog, { 
      store, 
      localVue,
    });

    const buttons = wrapper.findAll('.v-card__actions > .v-btn');
    expect(buttons.at(0).classes()).toContain('v-btn--text');
    expect(buttons.at(0).classes()).toContain('accent--text');
    expect(buttons.at(1).classes()).toContain('primary');
  })
})
