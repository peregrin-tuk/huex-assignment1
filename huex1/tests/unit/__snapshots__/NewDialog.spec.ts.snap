// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbutton should match snapshot 1`] = `
<v-card-stub loaderheight="4" tag="div">
  <v-card-title-stub>Create new board</v-card-title-stub>
  <v-card-text-stub>
    Create a new board? You can reopen this board by clicking on the <strong>Open</strong> button on the left side, or by saving and opening the URL at the top of your address bar.
  </v-card-text-stub>
  <v-card-actions-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-btn-stub color="accent" tag="button" activeclass="" text="true" type="button">Cancel</v-btn-stub>
    <v-btn-stub color="primary" tag="button" activeclass="" type="button">Create new board</v-btn-stub>
  </v-card-actions-stub>
</v-card-stub>
`;
